lib/cli-config.js

```diff
@@ -70,12 +72,16 @@ exports.getContent = function(config, directory) {
 
         if (ext === '.js' || ext === '.json') {
             content = require(configPath);
+        } else if (ext === '.yaml') {
+            content = yaml.safeLoad(fs.readFileSync(configPath, 'utf8'));
         } else {
-            content = JSON.parse(
-                stripJSONComments(
-                    fs.readFileSync(configPath, 'utf8')
-                )
-            );
+            data = fs.readFileSync(configPath, 'utf8');
+
+            try {
+                content = JSON.parse(stripJSONComments(data));
+            } catch (e) {
+                content = yaml.safeLoad(data);
```

Comment:
Understood. The biggest challenge is figuring out what the user's intention was with regards to format. We could write it:

```js
data = stripJSONComments(fs.readFileSync(configPath, 'utf8'));

try {
  content = yaml.safeLoad(data);
} catch (yamlError) {
  // not valid YAML, so by definition also not valid JSON
  if (data[0] === '{') {
    // assume the intention was JSON
    content = jsonlint.parse(data);
  } else {
    throw yamlError;
  }
}
```

Would that work for you? I could update it to be like that.